### DISCUSSION TITLE
How to get a click in GraphicsView?

### DISCUSSION THREAD

**Discussion by mcguu** [dotnet/maui#5049] [Q&A]
```
public class MyView:GraphicsView
{
    public MyView()
        {
             Drawable=new MyViewDrawable();
        }
}
```

How to get a click in MyView?

**Comment by nogginbox** [comment]
[GraphicsView](/dotnet/docs-maui/blob/main/docs/user-interface/controls/graphicsview.md) does not emit click events.

If you want a GraphicsView that can, then I think the easiest way to do this would be to use a Button, with a GraphicsView as its content. You can then get the click events for the button as users click on the content of the button.

**Reply by mcguu** [reply]
What do a Move ? I want to get the mouse's position? Button has it?

what 's more I want to understand a view-handler in a view how to do it?

**Reply by nogginbox** [reply]
I'm not sure if you can get access to touch or move events from inside Maui. I've not been able to find any move events or touch events in the source, but I might be missing something.

**Reply by nogginbox** [reply]
The documentation is getting better, but it's not all there. I often fall back to checking the source code or checking how things were done in Xamarin Forms, as they are often the same.

**Reply by mcguu** [reply]
@NogginBox  I wan to check the source code too but I do not understand it is there any in youtube about it?

**Reply by nogginbox** [reply]
I don't understand 99% of it.

I mainly just browse the [source code for the controls](/dotnet/maui/tree/main/src/Controls/src/Core) and check what properties and events are available and what each control inherits from.

**Reply by jsuarezruiz** [reply]
GraphicsView have several events to detect touch events (touch down, move, cancel etc). https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.graphicsview.startinteraction?view=net-maui-7.0

**Comment by mrrrTom** [comment]
https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/tap?view=net-maui-7.0
